src/celestia/qt/qtdeepskybrowser.cpp

```diff
@@ -184,8 +184,16 @@ void populateDsoVector(std::vector<DeepSkyObject*>& dsos,
             return;
         }
 
-        if (DeepSkyObject* dso = dsodb.getDSO(index); filter(dso))
+        DeepSkyObject* dso = dsodb.getDSO(index);
```

Comment:
```suggestion
        const DeepSkyObject* dso = dsodb.getDSO(index);
```
Not sure if it breaks something else, so please check

Reply:
Using `const` creates multiple compile errors on C++ vector usage. My version of C++ gives error message with potential suggestions then next line gives statement using suggestions could create ill-formed conversions. Best not to make change as compiler versions could react unexpectedly.